app/src/test/java/com/ikarabulut/GameDisplayTest.java

```diff
@@ -1,41 +1,112 @@
 package com.ikarabulut;
 
-import org.junit.jupiter.api.DisplayName;
-import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.*;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
 
 import static org.junit.jupiter.api.Assertions.*;
 
 class GameDisplayTest {
+
+    private final PrintStream standardOut = System.out;
+    private final ByteArrayOutputStream outputStreamCaptor = new ByteArrayOutputStream();
+    private static GameDisplay gameDisplay;
+
+    @BeforeAll
+    static void setByteArrayOutputStream() {
+        gameDisplay = new GameDisplay();
+    }
+
+    @BeforeEach
+    void setUp() {
+        System.setOut(new PrintStream(outputStreamCaptor));
+    }
+
+    @AfterEach
+    void tearDown() {
+        System.setOut(standardOut);
+    }
+
+    @AfterAll
+    static void tearByteArrayOutputStream() {
+        gameDisplay = null;
+    }
```

Comment:
@jramnani does this solve this issue [https://github.com/ikarabulut/JavaGameOfLife/pull/8#discussion_r877373384]

Reply:
The link to the comment is broken. But if it's the one I'm thinking about, "what happens when the tests fail and System.out is still set to a PrintStream?", then this doesn't solve it, but the `@AfterEach` does.

Reply:
Ok, so 
`    @AfterEach
    void tearDown() {
        System.setOut(standardOut);
    }`
Will set the System.out to System.out since that is what standardOut is set to in the global variable? Which resets it to its normal setup?

This @AfterAll has the wrong name, It should be to set the GameDisplay object reference to null.

Reply:
That is correct. You are saving the original location of `System.out` in the `standardOut` variable, replacing it within the test, then restoring it back to its original value.